File: n-folds.hs
    
    main = do

        putStrLn "thickness of the paper in mm:"
        t <- readLn
        let n1 =  t * 3.1415

        putStrLn "width of the paper in mm:"
        w <- readLn
        let n2 = w / n1

        let n3 = logBase 2 n2

        let n4 = 0.66667 * n3

        let n5 = 1 + n4

        let n6 = floor n5

        putStrLn ("number of possible folds are: " ++ show n6)

File: n-folds.py
#!/usr/bin/python

import math

pi = 3.1415

# input values of W and t from user

ui = input("enter value of t (thickness, gms, g/m2 of paper in milimeters): ")

s1 = ui * pi

ui = input("now enter value of W (width of square paper in milimeters): ")

s2 = ui / s1

#calculate log base 2

x = ui / s1
base = 2

logof = math.log(x, base)

#calculate n : number of folds

m1 = 0.66 * logof

m2 = m1 + 1

m3 = uconv = int(m2)

print "number of folds possible are: ", m3
